server/middleware/payments.coffee

```diff
@@ -5,11 +5,19 @@ parse = require '../commons/parse'
 
 all = wrap (req, res) ->
   throw new errors.Unauthorized('You must be an administrator.') unless req.user?.isAdmin()
-  query = if req.query.nofree then {amount: {$gt: 0}} else {}
-  if req.query.limit
-    payments = yield Payment.find(query, parse.getProjectFromReq(req)).limit(parse.getLimitFromReq(req)).lean()
-  else
-    payments = yield Payment.find(query, parse.getProjectFromReq(req)).lean()
+  query = {}
+  if req.query.nofree
+    query.amount = {$gt: 0}
+  if req.query.payPalResource
+    query['$or'] = [
+      {'payPal.transactions.related_resources.sale.id': req.query.payPalResource},
+      {'payPalSale.id': req.query.payPalResource},
+    ]
+  dbq = Payment.find(query, parse.getProjectFromReq(req))
+  dbq.limit(parse.getLimitFromReq(req))
```

Comment:
Will this limit break existing use cases that didn't have a limit applied?

Reply:
Yes, will change this to not be limited by default.